From 13fcfa01575417c3ccd30c183751dcb1e97653bf Mon Sep 17 00:00:00 2001
From: Xie XiuQi <xiexiuqi@huawei.com>
Date: Fri, 26 Feb 2021 20:21:11 +0800
Subject: [PATCH] arm64/mpam: add L3TALL & HHALL
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 13fcfa01575417c3ccd30c183751dcb1e97653bf


hulk inclusion
category: feature
feature: ARM MPAM support
bugzilla: 48265
CVE: NA

--------------------------------

Signed-off-by: Xie XiuQi <xiexiuqi@huawei.com>
Reviewed-by: Hanjun Guo <guohanjun@huawei.com>
Signed-off-by: Yang Yingliang <yangyingliang@huawei.com>
Signed-off-by: Wang ShaoBo <bobo.shaobowang@huawei.com>
Reviewed-by: Cheng Jian <cj.chengjian@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/arm64/kernel/mpam.c | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/arch/arm64/kernel/mpam.c b/arch/arm64/kernel/mpam.c
index d14693e139bb..9cc9b2a232a9 100644
--- a/arch/arm64/kernel/mpam.c
+++ b/arch/arm64/kernel/mpam.c
@@ -103,6 +103,13 @@ struct mpam_node mpam_node_all[] = {
 		.cpus_list              = "3",
 		.default_ctrl		= 0x7fff,
 	},
+	{
+		.name			= "L3TALL",
+		.type                   = MPAM_RESOURCE_CACHE,
+		.addr                   = 0x90B90000,
+		.cpus_list              = "0-3",
+		.default_ctrl		= 0x7fff,
+	},
 
 	/* P0 DIE 0: HHA0 */
 	{
@@ -119,6 +126,13 @@ struct mpam_node mpam_node_all[] = {
 		.addr                   = 0x90420000,
 		.cpus_list              = "0-3",
 	},
+
+	{
+		.name			= "HHALL",
+		.type                   = MPAM_RESOURCE_MC,
+		.addr                   = 0x90C10000,
+		.cpus_list              = "0-3",
+	},
 	/* other mpam nodes ... */
 };
 
@@ -1229,6 +1243,11 @@ static void mpam_domains_init(struct resctrl_resource *r)
 			r->mon_enabled
 			);
 
+		r->alloc_capable = 1;
+		r->alloc_enabled = 1;
+		r->mon_capable = 1;
+		r->mon_enabled = 1;
+
 		d->cpus_list = n->cpus_list;
 
 		d->ctrl_val = kmalloc_array(rr->num_partid, sizeof(*d->ctrl_val), GFP_KERNEL);
-- 
2.26.2

